analyzers/src/SonarAnalyzer.CSharp/Rules/AssertionArgsShouldBePassedInCorrectOrder.cs

```diff
@@ -18,63 +18,63 @@
+            var methodCall = (InvocationExpressionSyntax)c.Node;
+            if (!methodCall.Expression.IsKind(SyntaxKind.SimpleMemberAccessExpression)
+                || methodCall.ArgumentList.Arguments.Count < 2)
             {
-                var methodCall = (InvocationExpressionSyntax)c.Node;
-                if (!methodCall.Expression.IsKind(SyntaxKind.SimpleMemberAccessExpression)
-                    || methodCall.ArgumentList.Arguments.Count < 2)
-                {
-                    return;
-                }
+                return;
+            }
 
-                var firstArgument = methodCall.ArgumentList.Arguments[0];
-                var secondArgument = methodCall.ArgumentList.Arguments[1];
-                if (firstArgument.Expression is LiteralExpressionSyntax
-                    || secondArgument.Expression is not LiteralExpressionSyntax)
-                {
-                    return;
-                }
+            var argumentList = methodCall.ArgumentList;
+            var parameterLookup = new CSharpMethodParameterLookup(argumentList, c.SemanticModel);
+            parameterLookup.TryGetSyntax("expected", out var expected);
+            parameterLookup.TryGetSyntax("actual", out var actual);
+            if (expected.FirstOrDefault() is LiteralExpressionSyntax || actual.FirstOrDefault() is not LiteralExpressionSyntax)
+            {
+                return;
+            }
 
-                var methodCallExpression = (MemberAccessExpressionSyntax)methodCall.Expression;
+            var methodCallExpression = (MemberAccessExpressionSyntax)methodCall.Expression;
 
-                var methodKnownTypes = MethodsWithType.GetValueOrDefault(methodCallExpression.Name.Identifier.ValueText);
-                if (methodKnownTypes == null)
-                {
-                    return;
-                }
+            var methodKnownTypes = MethodsWithType.GetValueOrDefault(methodCallExpression.Name.Identifier.ValueText);
+            if (methodKnownTypes == null)
+            {
+                return;
+            }
 
-                var symbolInfo = c.SemanticModel.GetSymbolInfo(methodCallExpression.Expression).Symbol;
-                var isAnyTrackedAssertType = (symbolInfo as INamedTypeSymbol).IsAny(methodKnownTypes);
-                if (!isAnyTrackedAssertType)
-                {
-                    return;
-                }
+            var symbolInfo = c.SemanticModel.GetSymbolInfo(methodCallExpression.Expression).Symbol;
+            var isAnyTrackedAssertType = (symbolInfo as INamedTypeSymbol).IsAny(methodKnownTypes);
+            if (!isAnyTrackedAssertType)
+            {
+                return;
+            }
 
-                c.ReportIssue(Diagnostic.Create(Rule, firstArgument.CreateLocation(secondArgument)));
-            },
-            SyntaxKind.InvocationExpression);
-    }
+            c.ReportIssue(Diagnostic.Create(Rule, argumentList.Arguments[0].CreateLocation(argumentList.Arguments[1])));
```

Comment:
Can you try to rewrite this to the form we usually try to follow? It looks like this would be easier to read
```cs
if (xx
    && yy
    && zz)
{
    c.ReportIssue();
}
```